serving/ingress/pkg/reconciler/ingress/resources/route.go

```diff
@@ -111,6 +111,26 @@ func makeRoute(ci *networkingv1alpha1.Ingress, host string, rule networkingv1alp
 		return nil, ErrNoValidLoadbalancerDomain
 	}
 
+	terminationPolicy := routev1.InsecureEdgeTerminationPolicyAllow
+	if ci.Spec.HTTPOption == networkingv1alpha1.HTTPOptionRedirected {
+		terminationPolicy = routev1.InsecureEdgeTerminationPolicyRedirect
+	}
+
+	// TODO: Remove this annotation handling after serving 0.26+.
+	// Ingress configures the HTTPOption based on the annotation.
+	// https://github.com/knative/serving/commit/d9c1342b5761afdac88c563535885e37fae27c7e
+	if len(annotations) != 0 && annotations[networking.HTTPOptionAnnotationKey] != "" {
```

Comment:
```suggestion
	if annotations[networking.HTTPOptionAnnotationKey] != "" {
```

accessing nil and/or empty maps works gracefully. Only adding to them needs an explicit nil-check.